test_conformance/compiler/test_compiler_defines_for_extensions.cpp

```diff
@@ -42,6 +42,7 @@ const char *known_extensions[] = {
     "cl_khr_mipmap_image_writes",
     "cl_khr_srgb_image_writes",
     "cl_khr_subgroup_named_barrier",
+    "cl_khr_extended_async_copies",
```

Comment:
Should cl_khr_extended_async_copies have an extension macro defined in OpenCL C? I.e. should it be in the this uppermost "language extensions" part of the known_extensions array?  It's not clear to me from the extension spec whether or not it should.

Reply:
I've since been made aware that in the OpenCL Extension Specification § 1.2. Compiler Directives for Optional Extensions, the following is stated:

> Every extension which affects the OpenCL language semantics, syntax or adds built-in functions to the language must create a preprocessor `#define` that matches the extension name string. This `#define` would be available in the language if and only if the extension is supported on a given implementation.

There are three extensions for which the OpenCL Extension Specification explicitly mentions the preprocessor macro that needs to be defined, which may give the false impression that these extensions are special in that regard.  There is no such mention made for cl_khr_extended_async_copies.

Reply:
Right, I think that cl_khr_extended_async_copies should have an OpenCL C preprocessor define because it adds new built-in functions, so it's correct to include it in this section.

I don't have a very strong opinion whether we list the OpenCL C preprocessor define for all language extensions or whether the general statement in section 1.2 is sufficient, though I do agree listing it explicitly for some extensions but not others could be a cause for confusion.  Feel free to file an OpenCL-Docs issue if you'd like and we can discuss further.